### PR TITLE
Fix double copy

### PR DESCRIPTION
**Pull request by izgzhen**
Verified that with this change, https://github.com/snowleopard/hadrian/issues/323 can be resolved. But this is still a HACK because of https://github.com/snowleopard/hadrian/issues/318.

### CODE REVIEW

File: src/Rules/Install.hs [hunk 1]
```diff
@@ -200,15 +200,15 @@ installPackages = do
...
+                    quietly $ copyDirectoryContentsUntracked (Not excluded)
                                   installDistDir (installDistDir -/- "build")
 
                 pkgConf <- pkgConfFile context
                 need [ cabalFile, pkgConf ] -- TODO: check if need pkgConf
                 build $ Target context GhcCabalHsColour [cabalFile] []
 
                 -- HACK (#318): copy stuff back to the place favored by ghc-cabal
-                quietly $ copyDirectoryContents (Not excluded)
+                quietly $ copyDirectoryContentsUntracked (Not excluded)
```
Comment (snowleopard) [inline src/Rules/Install.hs]: It looks like when `isSpecified HsColour` holds, you'll do the copying twice?

Comment (izgzhen) [inline src/Rules/Install.hs]: Indeed!

File: src/Rules/Install.hs [hunk 2]
```diff
@@ -194,22 +194,16 @@ installPackages = do
...
                 pkgConf <- pkgConfFile context
                 need [ cabalFile, pkgConf ] -- TODO: check if need pkgConf
-                build $ Target context GhcCabalHsColour [cabalFile] []
 
                 -- HACK (#318): copy stuff back to the place favored by ghc-cabal
-                quietly $ copyDirectoryContents (Not excluded)
+                quietly $ copyDirectoryContentsUntracked (Not excluded)
                             installDistDir (installDistDir -/- "build")
+
+                build $ Target context GhcCabalHsColour [cabalFile] []
```
Comment (snowleopard) [inline src/Rules/Install.hs]: This line should be conditional on `whenM (isSpecified HsColour)`, right?

Comment (izgzhen) [inline src/Rules/Install.hs]: Right!

**Comment by izgzhen** [comment]
I've just verified that tracked copy is not really the source of lint error, while double copy probably is, which is fixed now.

**Comment by snowleopard** [comment]
Ah, great! I'll merge this once CI passes.

**Comment by snowleopard** [comment]
Not sure what happened with AppVeyor and CircleCI. I've restarted AppVeyor.

**Pull request merged**